picasso/src/main/java/com/squareup/picasso3/EventListener.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.picasso3
+
+import android.graphics.Bitmap
+import java.io.Closeable
+
+interface EventListener : Closeable {
+  fun cacheMaxSize(maxSize: Int)
+  fun cacheSize(size: Int)
+  fun cacheHit()
+  fun cacheMiss()
```

Comment:
These callbacks don't feel sufficiently general. Right now they're just an abstraction on stats essentially.

I would prefer to see something closer to OkHttp. We want to model the events that are happening: request enqueued, request started, request success, and request fail are probably good to start with. The success callback can inform cache hit/miss as a parameter and the final bitmap.

Reply:
Will address in a follow-up.